tests/datastore/test_base.py

```diff
@@ -54,11 +54,37 @@ def test_http_fs_parquet_with_params_as_df():
     data_item.as_df()
 
 
-def test_s3_fs_parquet_as_df():
+@pytest.mark.skipif(not is_pyarrow_17_or_later(), reason="Requires PyArrow 17+")
+def test_s3_fs_parquet_with_conflicts_as_df():
+    import mlrun.datastore
+
     data_item = mlrun.datastore.store_manager.object(
         "s3://aws-public-blockchain/v1.0/btc/blocks/date=2023-02-27/"
         "part-00000-7de4c87e-242f-4568-b5d7-aae4cc75e9ad-c000.snappy.parquet"
     )
+    try:
+        data_item.as_df()
+        # If this succeeds, the conflict may have been resolved
+        print("File read successfully - no schema conflict detected")
+    except mlrun.errors.MLRunValueError as e:
+        # This should now show our helpful message
+        error_msg = str(e)
+        print("Caught schema conflict with helpful message:")
+        print(error_msg)
+        # For testing, we can assert the message contains key information
+        assert "PYARROW SCHEMA CONFLICT DETECTED" in error_msg
+
+    except Exception as e:
+        # Other errors should still be raised normally
+        print(f"Other error (not schema conflict): {e}")
+        raise
```

Comment:
This is way too long and chatty...
```suggestion
    with pytest.raises(
        mlrun.errors.MLRunValueError,
        match="PYARROW SCHEMA CONFLICT DETECTED"
    ):
        data_item.as_df()
```